### PR TITLE
- added All Cities endpoint implementation

### PR DESCRIPTION
**Pull request by fortamt**
develop

## GitHub Board

**Issue link**

[Create new endpoint for obtains list of cities #574 ](https://github.com/ita-social-projects/dokazovi-be/issues/574 )

**Story link**

[Page overview #544](https://github.com/ita-social-projects/dokazovi-requirements/issues/544 )

## Code reviewers
- [ ] @teaFunny 
- [ ] @V-Kaidash 

## Summary of issue

- [x] Implement api endpoint contorller for getting all cities.
- [x] Implement api endpoint contorller for getting cities by region.
- [x] Implement api endpoint contorller for getting region by city.
- [x] Implement service for all above
- [x] Implement repository for all above
- [x] Add tests for all above.

## Summary of change

- [x] Implemented api endpoint contorller for getting all cities.
- [x] Implemented api endpoint contorller for getting cities by region.
- [x] Implemented api endpoint contorller for getting region by city.
- [x] Implemented service for all above
- [x] Implemented repository for all above
- [x] Added tests for all above.

### CODE REVIEW

File: src/main/java/com/softserveinc/dokazovi/controller/CityController.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+    @GetMapping
+    @ApiOperation("Get all cities")
+    public ResponseEntity<List<CityDTO>> getAllCities() {
+        return ResponseEntity
+                .status(HttpStatus.OK)
+                .body(cityService.findAllCities());
+    }
+
+    @PostMapping(CITIES_BY_REGION)
```
Comment (V-Kaidash) [inline src/main/java/com/softserveinc/dokazovi/controller/CityController.java:35]: Why did you use Post mapping? 
It seems this endpoint doesn't require body to be passed, hence you van use Get mapping.
Or did I miss something?

File: src/main/java/com/softserveinc/dokazovi/controller/RegionController.java [hunk 2]
```diff
@@ -36,4 +39,12 @@ public ResponseEntity<List<RegionDTO>> getAllRegions() {
 				.status(HttpStatus.OK)
 				.body(regionService.findAllRegions());
 	}
+
+	@PostMapping(REGION_BY_CITY)
```
Comment (V-Kaidash) [inline src/main/java/com/softserveinc/dokazovi/controller/RegionController.java:43]: It seems this endpoint doesn't require body to be passed, hence you van use Get mapping.
Or did I miss something?

**Review by V-Kaidash — CHANGES_REQUESTED** [review]
Please add some demo on this feature.
Thanks

File: src/main/java/com/softserveinc/dokazovi/controller/CityController.java (hunk 1 above)
Comment (fortamt) [reply src/main/java/com/softserveinc/dokazovi/controller/CityController.java:35]: My bad. I thought @PathVariable required @PostMapping too. Fixed

File: src/main/java/com/softserveinc/dokazovi/controller/RegionController.java (hunk 2 above)
Comment (fortamt) [reply src/main/java/com/softserveinc/dokazovi/controller/RegionController.java:43]: My bad. I thought @PathVariable required @PostMapping too. Fixed

**Comment by fortamt** [comment]
> Please add some demo on this feature. Thanks

https://drive.google.com/file/d/1HTS62XcuwPeRTC4_V1aZgEY3pOxvldno/view?usp=sharing

**Review by V-Kaidash — APPROVED** [review]

**Pull request merged**